nodewatcher/modules/devices/ubnt/edgerouter.py

```diff
@@ -0,0 +1,54 @@
+from django.utils.translation import ugettext_lazy as _
+
+from nodewatcher.core.generator.cgm import base as cgm_base, protocols as cgm_protocols, devices as cgm_devices
+
+
+class UBNTEdgerouterX(cgm_devices.DeviceBase):
+    """
+    UBNT EdgerouterX device descriptor.
+    """
+
+    identifier = 'ub-edgerouter-x'
+    name = "Edgerouter X"
+    manufacturer = "Ubiquiti"
+    url = 'http://www.ubnt.com/'
+    architecture = 'ramips_mt7621'
+    switches = [
+        cgm_devices.Switch(
+            'sw0', "Switch0",
+            ports=5,
+            cpu_port=6,
+            vlans=16,
+            configurable=True,
+            presets=[
+                cgm_devices.SwitchPreset('default', _("Default VLAN configuration"), vlans=[
+                    cgm_devices.SwitchVLANPreset(
+                        'lan0', "Lan0",
+                        vlan=1,
+                        ports=[0, 1, 2, 3, 4],
+                    )
+                ])
+            ]
+        )
+    ]
+    ports = [
+        cgm_devices.EthernetPort('wan0', "Wan0"),
+    ]
+    port_map = {
+        'openwrt': {
+            'sw0': cgm_devices.SwitchPortMap('switch0', vlans='eth0'),
```

Comment:
Port mapping seems incorrect. You have set static `eth0` for all VLANs and also `wan0`. If the switch actually supports VLANs, the mapping should be something like `eth0.{vlan}`. But you need to set this based on how the switch is configured in LEDE, please check that (maybe link to the section in LEDE source where defaults for this device are configured).

But especially you can't have both the switch and `wan0` on the same interface (e.g. `eth0`). If the WAN port is actually part of the switch, it needs to be configured as a switch VLAN preset, similar to the LAN port (with the correct port definitions of course).